### ISSUE TITLE
ivygapSE package submitted

### ISSUE THREAD

**Issue by vjcitn** [Bioconductor/Contributions#492]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/vjcitn/ivygapSE

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by mtmorgan** [comment]
Looks like there was a version bump but no build; probably due to bad web hook. Some troubleshooting [available](https://github.com/Bioconductor/Contributions#additional-actions).

**Comment by vjcitn** [comment]
are all the SPB hosts running?  i am only getting results for veracruz --
and it is full pass there.  webhook looks ok for ivygapSE and hmdbQuery
(has green checkmark for last delivery)

**Comment by mtmorgan** [comment]
Yes I think you're right that the problem is with the builders not  your package. Lori is working on this.

**Comment by mtmorgan** [comment]
DESCRIPTION

- ok, though brief

R

- the interactive function seems to be broken

    ```
    ivygapSE master$ R -e "ivygapSE::ivyGlimpse()"
    Bioconductor version 3.6 (BiocInstaller 1.27.5), ?biocLite for help
    > ivygapSE::ivyGlimpse()
    Error in (function (classes, fdef, mtable)  : 
      unable to find an inherited method for function 'metadata' for signature '"NULL"'
    Calls: <Anonymous> -> metadata -> <Anonymous>
    Execution halted
    ```
    
- Please provide some level of consistent indentation, e.g., ivyGlimpse.R: 24 is in the body of the function but not indented. Without this the code is hard to parse.

- Rationalize the definition of constants, e.g., why are there all of these?

    ```
    ivygapSE master$ grep -r "pi3k =" *
    inst/ivyGlimpse/server.R:pi3k = c(
    inst/ivyGlimpse/ui.R:pi3k = c(
    R/ivyGlimpse.R:pi3k = c(
    R/genesets.R:   pi3k = c(
    ```

vignettes

- remove the pdf (it is stale anyway).

man

- please provide additional detail on the provenance and pre-processing of ivySE.

- makeGeneSets.Rd:16. Provide a more extensive descirption of \value{}. Typo catalogy -> catalog


inst/

- I think more of your app code should be under R/, where it will be checked with R CMD check, etc. The constants should be under R/. I also think that functions like `renderPlot()` should be delegating the work to functions that are also under R/.

**Comment by vjcitn** [comment]
On Wed, Oct 4, 2017 at 1:01 PM, Martin Morgan <notifications@github.com>
wrote:

> DESCRIPTION
>
>    - ok, though brief
>
> R
>
>    -
>
>    the interactive function seems to be broken
>
>    ivygapSE master$ R -e "ivygapSE::ivyGlimpse()"
>    Bioconductor version 3.6 (BiocInstaller 1.27.5), ?biocLite for help
>    > ivygapSE::ivyGlimpse()
>    Error in (function (classes, fdef, mtable)  :
>      unable to find an inherited method for function 'metadata' for signature '"NULL"'
>    Calls: <Anonymous> -> metadata -> <Anonymous>
>    Execution halted
>
>
> This has been repaired.  A basic problem is the fact that data() does not
load an object
visible to following code.  Thus I use load(system.file ...)

>
>    -
>
>     -
>
>    Please provide some level of consistent indentation, e.g.,
>    ivyGlimpse.R: 24 is in the body of the function but not indented. Without
>    this the code is hard to parse.
>
> Addressed but could be better


>
>    -
>    -
>
>    Rationalize the definition of constants, e.g., why are there all of
>    these?
>
>    ivygapSE master$ grep -r "pi3k =" *
>    inst/ivyGlimpse/server.R:pi3k = c(
>    inst/ivyGlimpse/ui.R:pi3k = c(
>    R/ivyGlimpse.R:pi3k = c(
>    R/genesets.R:   pi3k = c(
>
>
> the inst/ivyGlimpse folder was a hack to get something simple for
shinyapps.io -- now there
is only a call to ivyGlimpse, with some setup, in app.R there


>
>    -
>
>
> vignettes
>
>    - remove the pdf (it is stale anyway).
>
> man
>
>    -
>
>    please provide additional detail on the provenance and pre-processing
>    of ivySE.
>
>
will address this and remaining in 0.99.4

**Comment by mtmorgan** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/vjcitn.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://docs.google.com/forms/d/e/1FAIpQLSdlTbNjsQJDp0BA480vo4tNufs0ziNyNmexegNZgNieIovbAA/viewform

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `biocLite(\"ivygapSE\")`. The package 'landing page' will be created at

https://bioconductor.org/packages/ivygapSE

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.
